python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -239,6 +239,52 @@ def test_timestamps_notimezone_nulls(self):
             })
         self._check_pandas_roundtrip(df, timestamps_to_ms=False)
 
+    def test_timestamps_with_timezone_no_nulls(self):
+        df = pd.DataFrame({
+            'datetime64': np.array([
+                '2007-07-13T01:23:34.123',
+                '2006-01-13T12:34:56.432',
+                '2010-08-13T05:46:57.437'],
+                dtype='datetime64[ms]')
+            })
+        df_est = df['datetime64'].dt.tz_localize('UTC').dt.tz_convert('US/Eastern').to_frame()
```

Comment:
From my comment above, you can look at `x.values.view(np.int64)` before and after `tz_convert('US/Eastern')` to check that the internal values are not modified. If you localize with a non-UTC timestamp, that actually changes the values